Tests/varLib/instancer_test.py

```diff
@@ -1916,6 +1917,212 @@ def test_normalizeAxisLimits_missing_from_fvar(varfont):
+            name="Weight",
+            values=[
+                dict(value=400, name='Regular'),
+            ],
+        ),
+        dict(
+            tag="wdth",
+            name="Width",
+            values=[
+                dict(value=75, name="Condensed"),
+            ]
+        )
+    ]
+    buildStatTable(varfont, axes)
+    instancer.updateNameTable(varfont, {"wdth": 75, "wght": 400})
+    names = _get_name_records(varfont)
+    assert names[(17, 3, 1, 0x409)] == "Regular Condensed"
+
+    # Swap the axes so the names get swapped
+    axes[0], axes[1] = axes[1], axes[0]
+
+    buildStatTable(varfont, axes)
+    instancer.updateNameTable(varfont, {"wdth": 75, "wght": 400})
+    names = _get_name_records(varfont)
+    assert names[(17, 3, 1, 0x409)] == "Condensed Regular"
+
+
+def test_updateNameTable_with_multilingual_names(varfont):
+    name = varfont["name"]
+    name.setName("Test Variable Font", 1, 3, 1, 0x405)
+    name.setName("Normal", 2, 3, 1, 0x405)
+    name.setName("Normal", 261, 3, 1, 0x405) # nameID 261=Regular STAT entry
+    name.setName("Negreta",266, 3, 1, 0x405) # nameID 266=Black STAT entry
+    name.setName("Zhuštěné", 279, 3, 1, 0x405) # nameID 279=Condensed STAT entry
+
+    # Regular | Normal
+    instancer.updateNameTable(varfont, {"wght": 400})
+    names = _get_name_records(varfont)
+    assert names[(1, 3, 1, 0x405)] == "Test Variable Font"
+    assert names[(2, 3, 1, 0x405)] == "Normal"
+    assert (3, 3, 1, 0x405) not in names
+    assert (16, 3, 1, 0x405) not in names
+    assert (17, 3, 1, 0x405) not in names
+
+    # Black | Negreta
+    instancer.updateNameTable(varfont, {"wght": 900})
+    names = _get_name_records(varfont)
+    assert names[(1, 3, 1, 0x405)] == "Test Variable Font Negreta"
+    assert names[(2, 3, 1, 0x405)] == "Normal"
+    assert (3, 3, 1, 0x405) not in names
+    assert names[(16, 3, 1, 0x405)] == "Test Variable Font"
+    assert names[(17, 3, 1, 0x405)] == "Negreta"
+
+    # Black Condensed | Negreta Zhuštěné
+    instancer.updateNameTable(varfont, {"wdth": 79, "wght": 900})
+    names = _get_name_records(varfont)
+    assert names[(1, 3, 1, 0x405)] == "Test Variable Font Negreta Zhuštěné"
+    assert names[(2, 3, 1, 0x405)] == "Normal"
+    assert (3, 3, 1, 0x405) not in names
+    assert names[(16, 3, 1, 0x405)] == "Test Variable Font"
+    assert names[(17, 3, 1, 0x405)] == "Negreta Zhuštěné"
+
+
+def test_updateNametable_partial(varfont):
+    instancer.updateNameTable(varfont, {"wdth": 79, "wght": (400, 900)})
+    names = _get_name_records(varfont)
+    assert names[(1, 3, 1, 0x409)] == "Test Variable Font Condensed"
+    assert names[(2, 3, 1, 0x409)] == "Regular"
+    assert (3, 3, 1, 0x405) not in names
+    assert names[(16, 3, 1, 0x409)] == "Test Variable Font"
+    assert names[(17, 3, 1, 0x409)] == "Condensed"
+
+
+def test_updateNameTable_missing_axisValues(varfont):
+    with pytest.raises(ValueError, match="Cannot find Axis Values for axes \[wght\]"):
```

Comment:
maybe the error message should specify for which input location it cannot find axis values for an axis. Just saying "Cannot find Axis Values for axes [...]" is too broad, seems to imply there are none for that axis. It might be there are some, only that the ones requested are not there